Documentation/guides/cpp_cmake.rst

```diff
@@ -0,0 +1,287 @@
+    set(CMAKE_OBJDUMP       ${COMPILER_PREFIX}objdump)
+    set(CMAKE_SIZE          ${COMPILER_PREFIX}size)
+
+    set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+    set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+    set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+    set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+
+    set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+
+    set(AC_HW_FLAGS         "-mcpu=cortex-m4 -mthumb -mfloat-abi=soft ")
+    set(AC_HW_FLAGS         "${AC_HW_FLAGS} -isystem ${NUTTX_PATH}/include")
+    set(AC_HW_FLAGS         "${AC_HW_FLAGS} -pipe -D__NuttX__")
+
+    set(AC_LINKER_FLAGS     "--entry=__start -nostartfiles -nodefaultlibs -T${MCU_LINKER_SCRIPT}")
+
+* hellocpp/src/CMakeLists.txt
+
+.. code-block:: cmake
+    set(HEADER_FILES
+            HelloWorld.h
+    )
+
+    set(SOURCE_FILES
+            HelloWorld.cpp
+    )
+
+    link_directories(${EXE_NAME} ${NUTTX_PATH}/libs)
+
+    add_executable(${EXE_NAME} ${SOURCE_FILES} main.cpp ${HEADER_FILES})
+
+    add_custom_command(
+            TARGET ${EXE_NAME}
+            POST_BUILD
+            COMMAND ${CMAKE_OBJCOPY} ARGS -S -O binary ${CMAKE_BINARY_DIR}/${EXE_NAME}.elf ${CMAKE_BINARY_DIR}/${EXE_NAME}.bin
+    )
+
+    target_link_libraries(${EXE_NAME} --start-group)
+
+    target_link_libraries(${EXE_NAME} sched)
+    target_link_libraries(${EXE_NAME} drivers)
+    target_link_libraries(${EXE_NAME} boards)
+    target_link_libraries(${EXE_NAME} c)
+    target_link_libraries(${EXE_NAME} mm)
+    target_link_libraries(${EXE_NAME} arch)
+    target_link_libraries(${EXE_NAME} xx)
+    target_link_libraries(${EXE_NAME} apps)
+    target_link_libraries(${EXE_NAME} fs)
+    target_link_libraries(${EXE_NAME} binfmt)
+    target_link_libraries(${EXE_NAME} board)
+    target_link_libraries(${EXE_NAME} gcc)
+    target_link_libraries(${EXE_NAME} supc++)
+
+    target_link_libraries(${EXE_NAME} --end-group)
+
+* hellocpp/src/main.cpp
+
+.. code-block:: c++
+
+    #include <stdio.h>
+    #include "HelloWorld.h"
+    #include "Log.h"
+    #include <nuttx/config.h>
+    #include <iostream>
+
+    extern "C"
+    {
+            int phigw_main(void)
+            {
+                    Log::print("PHI: Exemplo em CMake");
+
+                    CHelloWorld *pHelloWorld = new CHelloWorld();
+                    pHelloWorld->HelloWorld();
+
+                    CHelloWorld helloWorld;
+                    helloWorld.HelloWorld();
+
+                    delete pHelloWorld;
+                    return 0;
+            }
+    }
+
+* hellocpp/src/HelloWorld.h
+
+.. code-block:: c++
+
+    #ifndef HELLOWORLD_H_
+    #define HELLOWORLD_H_
+
+    #include "nuttx/config.h"
+
+    class CHelloWorld
+    {
+            public:
+                    CHelloWorld();
+                    ~CHelloWorld();
+                    bool HelloWorld(void);
+            private:
+                    int mSecret;
+    };
+
+    #endif
+
+* hellocpp/src/HelloWorld.cpp
+
+.. code-block:: c++
+
+    #include <stdio.h>
```

Comment:
```suggestion
    #include <cstdio>
```
nit: Since this is a C++ example, how about sticking with the functions from the standard library?